test(cypress): add a new tests to verify secrets are mounted properly

lorenzo-cavazzi opened the pull request:
This PR adds a new acceptance test for the new User Secrets feature

/deploy

leafty commented:
I'm not sure a feature dubbed as "alpha" should have an acceptance test 🤔

lokijuhy commented on CHANGELOG.rst:
```diff
@@ -1,8 +1,21 @@
+sessions.
+
+
+**🌟 New Features**
```
We are missing the Use-Facing Changes section header here

lokijuhy commented on CHANGELOG.rst:
```diff
@@ -1,8 +1,21 @@
+0.52.0
 ------
 
+Renku ``0.52.0`` contains a new secrets-storage service that allows users to store and use secrets in 
+sessions.
```
```suggestion
Renku ``0.52.0`` introduces a new feature to save and use secrets in RenkuLab sessions. For example, save your credentials for a database or external computing resource in RenkuLab to access those external systems from a RenkuLab session. Save secrets via the new User Secrets page in the account drop down, and choose which secrets to mount in a session on the Start with Options page. 
```

lokijuhy replied:
Can we also include a link to the docs?

lokijuhy reviewed (changes requested):
I don't know if this is an appropriate PR to leave Changeling comments, since I didn't get to comment on the initial PR ?

lorenzo-cavazzi commented:
> I don't know if this is an appropriate PR to leave Changeling comments, since I didn't get to comment on the initial PR ?

Sorry @lokijuhy the `Files changed` tab was showing a bunch of changes totally unrelated to the PR (now fixed -- I rebased to the release branch).

Do you mind moving the release/feature changes request to the release PR?

leafty commented on cypress-tests/cypress/e2e/useSession.cy.ts:
```diff
@@ -398,9 +398,11 @@ describe("Basic public project functionality", () => {
       .contains("Start Session")
-      .should("exist")
+      .should("be.visible")
+      .should("be.enabled")
```
This test is not useful if you `.click()` on a button. Cypress fails if you try to click on a `disabled` element (unless you use the `{ force: true }` option).

```suggestion
```

lorenzo-cavazzi replied:
Doesn't Cypress wait until the button is enabled when I add `.should("be.enabled")`?

leafty replied:
Cypress already waits for the button to be enabled when using `.click()`. That line does not add anything new.

lokijuhy commented:
> > I don't know if this is an appropriate PR to leave Changeling comments, since I didn't get to comment on the initial PR ?
> 
> Sorry @lokijuhy the `Files changed` tab was showing a bunch of changes totally unrelated to the PR (now fixed -- I rebased to the release branch).
> 
> Do you mind moving the release/feature change requests to the release PR?

Ok, will do

leafty commented:
@lorenzo-cavazzi the test seems to have failed in the last run

rokroskar commented:
That test seems to be somewhat flaky recently

leafty commented:
It's the new test from this PR! So it should be check if this is because there is an issue with user secrets in this PR's deployment or if the test is not robust.

lorenzo-cavazzi commented:
Yes, we should not merge this "as is". Somehow, the test fails much more frequently than it should. From the videos, it seems that starting the session takes forever, but when doing it manually it's almost always quicker.
My plan is to:
1. Further investigate the problem
2. Modify the `useSession` test since it's currently the slowest and it can be split into multiple tests; ideally, some of the changes from the "Start with options" page can also be combined, no need to start too many separate sessions since most of the time goes there

olevski commented:
@lorenzo-cavazzi please target release-0.53.x
